torch/csrc/distributed/c10d/symm_mem/nccl_extension.cuh

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <c10/macros/Macros.h>
+#include <ATen/ATen.h>
+
+#ifdef USE_C10D_NCCL
+#include <nccl.h>
+#include <torch/csrc/cuda/nccl.h>
+#if NCCL_VERSION_CODE >= NCCL_VERSION(2, 28, 0)
+#define NCCL_HAS_SYMMEM_SUPPORT
+#endif
```

Comment:
This will entirely disable NCCL SymmMem for builds with NCCL 2.27? torch 2.10 CU12 build still uses 2.27.

Reply:
I really don't like this idea of multiple macros, but yes we can do that.